spark/ingestion/src/main/scala/feast/ingestion/StreamingPipeline.scala

```diff
@@ -92,8 +94,18 @@ object StreamingPipeline extends BasePipeline with Serializable {
 
         config.deadLetterPath match {
           case Some(path) =>
+            implicit def rowEncoder: Encoder[Row] = RowEncoder(projected.schema)
+
             batchDF
               .filter(!validator.checkAll)
+              .mapPartitions(iter => {
+                val res = iter
+                  .map(row => {
```

Comment:
Do we really need to iterate over all rows instead of just increment metric once by `iter.length`?